src/libstore/gc.cc

```diff
@@ -619,6 +619,18 @@ void LocalStore::collectGarbage(const GCOptions & options, GCResults & results)
         Path path = storeDir + "/" + std::string(baseName);
         Path realPath = realStoreDir + "/" + std::string(baseName);
 
+        /* There may be temp directories in the store that are still in use
+           by another process. We need to be sure that we can acquire an
+           exclusive lock before deleting them. */
+        AutoCloseFD tmpDirFd;
+        if (baseName.rfind("add-", 0) == 0) {
```

Comment:
Any reason the `AutoCloseFD` is declared outside of the block? Doesn't change a lot, but that make Nix create one instance for each store path rather than just the ones that match the pattern

```suggestion
        if (baseName.rfind("add-", 0) == 0) {
                AutoCloseFD tmpDirFd;
```